apollo-common/src/main/java/com/ctrip/framework/apollo/common/utils/ConfigValidator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.common.utils;
+
+import com.ctrip.framework.apollo.core.utils.StringUtils;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.util.Assert;
+
+/**
+ * Config Validator
+ *
+ * @author Lv Lifeng
+ */
+public class ConfigValidator {
+
+  private final ConfigurableApplicationContext applicationContext;
+
+  public ConfigValidator(ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+    checkDataBaseName();
+  }
+
+  private void checkDataBaseName() {
+    String checkFlag = applicationContext.getEnvironment().getProperty("config.check.databasename");
```

Comment:
Shall we change the property name to `apollo.check.database-name`?
And is it better to use the `ConditionalOnProperty` mechanism? e.g.

```java
@ConditionalOnProperty(value = {"apollo.check.database-name"}, havingValue = "true", matchIfMissing = true)
```

Reply:
OK, I will revise it in this way.

Reply:
I want to add logic to check the database name before creating the data source.
If use the `@ConditionalOnProperty` to deal with it, there will be a problem. There is a dependency in admin service,
`adminServiceAuthenticationFilter` -> `bizConfig` -> `bizDBPropertySource` -> `serverConfigRepository` ->  `serverConfig` ->  `entityManager` -> `dataSourceScriptDatabaseInitializer` -> `dataSource`, when the servlet is initialized, the filter will be initialized first,this will cause the data source to initialize, so unable to add name check before data source initialization. it seems meaningless if the name check is added after the data source is initialized，or is it the wrong way I handled it?